source/solution_deploy/lib/__snapshots__/member-stack.test.ts.snap

```diff
@@ -339,11 +339,11 @@ exports[`member stack snapshot matches 1`] = `
     },
     "NestedStackFactoryGatePlaybookMemberStackCIS120E08EFB8B": {
       "Metadata": {
-        "PlaybookMemberStackAFSBPReady": {
+        "PlaybookMemberStackAFSBPNestedStackPlaybookMemberStackAFSBPNestedStackResourceE0E577A8Ready": {
```

Comment:
If these logical IDs change, will that cause problems on stack update? Should we override them to the old values?

Reply:
It's possible, I'm not too sure on why these logical IDs are changing here to be completely honest with you. I'll investigate, but I think overriding them to the old values makes sense.

Reply:
Having just noticed this in instance scheduler as well, my hunch is that CDK adds a hash of something (input params?) to the logical id to ensure cfn updates things when they change. No idea if this could present a deployment problem or not though (my hunch would be no? but this last week hasn't been too kind to such hunches)

Reply:
Yes, at least for NestedStacks that is the behavior. This has been addressed by overriding the logical ids in the commit below, just to make sure we keep those static.

Reply:
@CrypticCabub The [logical ID of a resource](https://docs.aws.amazon.com/cdk/v2/guide/identifiers.html) is your assigned ID concatenated to the path of the resource stripped of separator characters. The path is a directory-like representation of the scope of a resource. CDK also concatenates a hash of the path to disambiguate paths that are distinct but resolve to the same string when separators are removed: `A/B/C` -> `ABC` and `A/BC` -> `ABC`. Examples:

```ts
class MyStack extends Stack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new Bucket(this, 'MyBucket'); // logicalId is 'MyBucket' + hash of 'MyBucket'
    new MyConstruct(this, 'MyConstruct'); // logicalId is 'MyConstruct' + hash of 'MyConstruct'
  }
}

class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new Bucket(this, 'MyOtherBucket'); // logicalId is 'MyConstructMyBucket' + hash of 'MyConstruct/MyBucket'
  }
} 
```

You need to care about the exact logical IDs of your resources because they are what controls CFN's update behavior. CFN will update a resource if the new template has a resource with the same logical ID (and resource type). If you change the logical ID, even by only renaming the higher level construct, CFN will create a new set of resources and then delete the old ones. If you don't care about update, this doesn't matter. If you use no static names, this will work but will probably take a little longer than update. If you have any static naming like in this project, you have to be very careful - CFN will try to create new resources with the same physical IDs (actual resource names), which will fail.

Reply:
thanks @hearde! very useful breakdown.